### ISSUE TITLE
Missmatch namespace between CLI and Webpack plugin

### ISSUE THREAD

**Issue by idoros** [wix/stylable#603]
When generating the namespace for a stylesheet there is a difference between the `@stylable/cli` and the `@stylable/webpack-plugin`. It is important to be able to generate identical namespaces in order to have proper server side rendering or static site generation that can later be matched by the CSS bundle.

The reason for this is that the CLI generates the namespace using the [default processNameSpace](https://github.com/wix/stylable/blob/cba96cde93af3871801dcc8c677c5679cfedcd51/packages/core/src/stylable-processor.ts#L589) while the webpack-plugin [passes](https://github.com/wix/stylable/blob/cba96cde93af3871801dcc8c677c5679cfedcd51/packages/webpack-plugin/src/stylable-webpack-plugin.ts#L2) the [`@stylable/node` resolver](https://github.com/wix/stylable/blob/cba96cde93af3871801dcc8c677c5679cfedcd51/packages/node/src/resolve-namespace.ts) which is using the package name for persistency.

I would prefer to have persistent class names across dev machines and CI _(and I can imagine Wix would need that as well)_, but currently the only way to workaround this is to pass the same simple `processNamespace` to the webpack-plugin and lose the persistent names:

```js
/*webpack.config.js*/
const { StylableWebpackPlugin } = require("@stylable/webpack-plugin")
const { processNamespace } = require("@stylable/core")

module.exports = {
    plugins: [
        new StylableWebpackPlugin({
            // override with the original simple namespace
            resolveNamespace: processNamespace,

            // Notice also that in order to have server/static rendering then some
            // optimizations cannot be enabled because the CLI doesn't know how to handle it. 
            // But that is a problem for another time.
            optimize: {
                shortNamespaces: false,
                classNameOptimizations: false
            },
    }),
}
```

**Comment by idoros** [comment]
BTW, if the CLI would allow for a custom namespace resolver, like the `resolveNamespace` of the webpack-plugin, then a utility can be made that generates the minified namespace while packging and then later uses that information to set the same namespace when generating the JS modules with the CLI.

**Comment by tomrav** [comment]
Resolved in #622, made the `webpack-plugin` and `CLI` default namespace resolvers match, as well as added the requested flag `--namespaceResolver` or `--nsr` for short..
